src/handlers/expenditures/expenditurePayoutClaimed.ts

```diff
@@ -1,7 +1,74 @@
 import { ContractEvent } from '~types';
+import { getExpenditureDatabaseId, output, toNumber, verbose } from '~utils';
+
+import { getExpenditureFromDB } from './helpers';
+import {
+  ExpenditurePayout,
+  UpdateExpenditureDocument,
+  UpdateExpenditureMutation,
+  UpdateExpenditureMutationVariables,
+} from '~graphql';
+import { mutate } from '~amplifyClient';
 
 export default async (event: ContractEvent): Promise<void> => {
-  // @TODO: Implement
-  console.log(event);
-  return;
+  const { contractAddress: colonyAddress } = event;
+  const { id: expenditureId, slot, token: tokenAddress } = event.args;
+  const convertedExpenditureId = toNumber(expenditureId);
+  const convertedSlotId = toNumber(slot);
+  const databaseId = getExpenditureDatabaseId(colonyAddress, expenditureId);
+
+  const expenditure = await getExpenditureFromDB(databaseId);
+  if (!expenditure) {
+    output(
+      `Could not find expenditure with ID: ${convertedExpenditureId} and colony address: ${colonyAddress} in the db. This is a bug and needs investigating.`,
+    );
+    return;
+  }
+
+  const existingSlotIndex = expenditure.slots.findIndex(
+    (slot) => slot.id === convertedSlotId,
+  );
+  const existingSlot = expenditure.slots[existingSlotIndex];
```

Comment:
Think you could just use `.find` here and below

Reply:
I need the index though - it's referred to in lines 50 and 58